src/Model/SoftDeletesTrait.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Mongolid\Model;
+
+use DateTime;
+use MongoDB\BSON\UTCDateTime;
+use Mongolid\Cursor\CursorInterface;
+use Mongolid\Util\QueryBuilder;
+
+trait SoftDeletesTrait
+{
+    public bool $enabledSoftDeletes = true;
+    public bool $forceDelete = false;
+
+    public function isTrashed(): bool
+    {
+        return !is_null($this->{self::getDeletedAtColumn()});
+    }
+
+    public function restore(): bool
+    {
+        $collumn = self::getDeletedAtColumn();
+
+        if (!$this->isTrashed()) {
+            return false;
+        }
+
+        unset($this->{$collumn});
+
+        return $this->execute('save');
```

Comment:
```suggestion
        return $this->update();
```
We could use update method here?

Reply:
Yes I tested here and the suggestion worked